### ISSUE TITLE
Kodi wakes up CEC-enabled TV 15 seconds after TV went into suspended mode.

### ISSUE THREAD

**Issue by mkuthan** [xbmc/xbmc#20510]
### Describe the bug

Kodi wakes up CEC-enabled TV without any manual operation. After screen saver / power saving timeout passes TV goes again into suspend mode and ~15 seconds later Kodi wakes up TV again. Again and again ...

## Expected Behavior
Suspended TV stays suspended until I activate Kodi explicitly (eg. via remote controller).

### To Reproduce
Perhaps not possible without my TV model and Raspberry Pi configuration :(

### Debuglog
When TV is "on" command `cec-ctl -m -d /dev/cec0` shows:

```
Driver Info:
	Driver Name                : vc4_hdmi
	Adapter Name               : vc4
	Capabilities               : 0x0000011e
		Logical Addresses
		Transmit
		Passthrough
		Remote Control Support
		Connector Info
	Driver version             : 5.10.63
	Available Logical Addresses: 1
	DRM Connector Info         : card 1, connector 32
	Physical Address           : 1.0.0.0
	Logical Address Mask       : 0x0000
	CEC Version                : 2.0
	OSD Name                   : ''
	Logical Addresses          : 0


Initial Event: State Change: PA: 1.0.0.0, LA mask: 0x0000, Conn Info: yes
```

When I turn off TV into suspend mode, after ~15 seconds the cec-ctl shows 3 lines at the same time:

```
(warn: State Change events were lost)
Event: State Change: PA: f.f.f.f, LA mask: 0x0000, Conn Info: yes

Event: State Change: PA: 1.0.0.0, LA mask: 0x0000, Conn Info: yes
```

Immediatelly after that Kodi wakes up my TV. 

### Your Environment
 - Rasperry PI 4
 - Linux: 5.10.63-v8+ (arm64)
 - Debian Bullseye
 - Kodi version: 19.3 (compiled from sources)
 - KMS video driver
 
 For FKMS driver CEC works perfectly, unfortunatelly FKMS does not register /dev/cecX devices so I'm unable to compare CEC messages.

Feel free to tell me what else I could check, I'll be glad to help.

**Comment by Langelus** [comment]
I experience the same thing with RPi 4, Denon AVR-1100W and Philips TV

**Comment by ErnstNeger** [comment]
Same here, will report the models later, also a Denon AVR, a Philips TV and a Raspberry Pi (3).

**Comment by Langelus** [comment]
Seems like a couple of people on the LibreELEC forum experience this as well. Found a link for some sort of patch that should possibly resolve it(?)

https://github.com/popcornmix/linux/commit/b5c047738ca72ac345820a4573b576a466584234

**Comment by popcornmix** [comment]
There is also a potential test build [posted here](https://forum.libreelec.tv/thread/24783-tv-avr-turns-back-on-right-after-turning-them-off/?postID=165969#post165969) but as no one has tested it, it won't be included in future builds.

**Comment by malard** [comment]
is this still an open issue?

**Comment by popcornmix** [comment]
Yes. Most users who hit it, workaround the issue with vc4.force_hotplug=1.

But I believe the real fix should be in libcec.

The workaround prevents the kernel from ever seeing a hotplug change.
That stops libcec from reacting to change and hides the problem.

**Comment by malard** [comment]
Okay, but then its not a libcec issue, because if the hotplug is flapping then libcec is responding according to the spec?

**Comment by popcornmix** [comment]
If you have a TV and AVR then switching off either one may result in a changed edid (as audio capabilities have changed).
To signal that you will get a hotplug deassert/assert. That is following hdmi spec.

The problem is when you turn off either TV or AVR, libcec sees the deassert/assert of hotplug and switches the TV back on.
That is undesirable (and basically makes switching TV off impossible without unplugging it from hdmi).
